src/esp/gfx/PBRShader.h

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_GFX_PBRSHADER_H_
+#define ESP_GFX_PBRSHADER_H_
+
+#include <memory>
+#include <unordered_set>
+#include <vector>
+
+#include <Corrade/Containers/EnumSet.h>
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/Trade/LightData.h>
+
+#include "esp/core/esp.h"
+
+namespace esp {
+
+namespace gfx {
+
+class PBRShader : public Magnum::GL::AbstractShaderProgram {
+ public:
+  /**
+   * @brief vertex positions
+   */
+  typedef Magnum::GL::Attribute<0, Magnum::Vector3> Position;
```

Comment:
I'd instead use this, so you are always sure that the same attribute is used here and in magnum's builtin shaders. Because if you get out of sync, things like `MeshTools::compile()` won't work anymore. [Details here.](https://doc.magnum.graphics/magnum/structMagnum_1_1Shaders_1_1Generic.html#Shaders-Generic-custom)

```suggestion
  typedef Magnum::Shaders::Generic3D::Position Position;
```

Same for other attribs below.